src/jit-analyze/jit-analyze.cs

```diff
@@ -121,18 +158,205 @@ public int GetHashCode(FileInfo fi)
             }
         }
 
+        public abstract class Metric
+        {
+            public virtual string Name { get; }
+            public virtual string DisplayName { get; }
+            public virtual string Unit { get; }
+            public virtual bool LowerIsBetter { get; }
+            public abstract Metric Clone();
+            public abstract string ValueString { get; }
+            public double Value { get; set; }
+
+            public void Add(Metric m)
+            {
+                Value += m.Value;
+            }
+
+            public void Sub(Metric m)
+            {
+                Value -= m.Value;
+            }
+
+            public void SetValueFrom(Metric m)
+            {
+                Value = m.Value;
+            }
+        }
+
+        public class CodeSizeMetric : Metric
+        {
+            public override string Name => "CodeSize";
+            public override string DisplayName => "Code Size";
+            public override string Unit => "bytes";
+            public override bool LowerIsBetter => true;
+            public override Metric Clone() => new CodeSizeMetric();
+            public override string ValueString => $"{Value}";
+        }
+
+        public class PrologSizeMetric : Metric
+        {
```

Comment:
I'm not sure if we care about PrologSizes, in my change I just removed it.

Reply:
Seems harmless to keep it, though. I expect going forward we will have lots of metrics.